src/power_grid_model_io/utils/filters.py

```diff
@@ -0,0 +1,151 @@
+# SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model IO project <dynamic.grid.calculation@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+"""
+These functions can be used in the mapping files to apply functions to tabular data
+"""
+
+import math
+import re
+from typing import Any, Optional, Tuple, TypeVar, cast
+
+import numpy as np
+from power_grid_model import WindingType
+
+CONNECTION_PATTERN = re.compile(r"(Y|YN|D|Z|ZN)(y|yn|d|z|zn)(\d|1[0-2])")
+
+"""
+TODO: Implement Z winding
+"""
+WINDING_TYPES = {
+    "Y": WindingType.wye,
+    "YN": WindingType.wye_n,
+    "D": WindingType.delta,
+    "Z": WindingType.wye,
+    "ZN": WindingType.wye_n,
+}
+
+T = TypeVar("T")
+
+
+def relative_no_load_current(i: float, p: float, s_nom: float, u_nom: float) -> float:
```

Comment:
@bramstoeller  since the conversion performance will be essential (for pandapower). We need to have vectorized version of all these filter functions. The input/output should be numpy arrays.

Reply:
I agree, partially.

My rationale was: Create a way to define functions in an understandable way. These mapping functions are much easier to read and understand than vectorized function. Currently we didn't have any efficiency constraints, but indeed, they'll come up in the near future. Let's create new stories for it.

Peter and I do have some ideas about efficiency (for pandapower), let's discuss it in a refinement meeting.